src/vignette/util/image_response.clj

```diff
@@ -38,32 +43,39 @@
 
 (defn create-image-response
   ([image image-map]
-   (-> (response (->response-object image))
-       (header "Content-Type" (content-type image))
-       (header "Content-Length" (content-length image))
-       (header "ETag" (etag image))
-       (header "X-Thumbnailer" "Vignette")
-       (add-content-disposition-header image-map)
-       (add-surrogate-header image-map)))
+   (as-> (response (->response-object image)) resp
+         (when-header-val resp "Content-Type" (content-type image))
+         (when-header-val resp "Content-Length" (content-length image))
+         (when-header-val resp "ETag" (etag image))
+         (header resp "X-Thumbnailer" "Vignette")
+         (add-content-disposition-header resp image-map image)
+         (add-surrogate-header resp image-map)))
   ([image]
     (create-image-response image nil)))
 
 (defn create-head-response
   [image-map]
   (-> (response "")
-       ;(header "Content-Type" (content-type image))
        (header "X-Thumbnailer" "Vignette")
        (add-content-disposition-header image-map)
        (add-surrogate-header image-map)))
 
+(defn- base-filename [image-map object]
```

Comment:
filename is taken from the stored object protocol if its not provided through image-map.
While I agree that current wikia image-map needs to be rethought I don't necessarily think that polymorphism will be the solution especially to this particualr case. 

Right now if image-map provides the filename its used here if not then stored object is used as sort of fallback.

I thought about removing filename from image map and rely only on storage object but I would like to make limit the scope of single PR of such core component that is Vignette.